From: Dave Martin <Dave.Martin@arm.com>
Date: Tue, 25 Jul 2017 12:52:41 +0100
Subject: arm64: sysreg: Fix unprotected macro argmuent in write_sysreg
Git-commit: d0153c7ff9226535a51e6a81f61656c9500957f4
Patch-mainline: v4.13-rc3
References: bsc#1068032

write_sysreg() may misparse the value argument because it is used
without parentheses to protect it.

This patch adds the ( ) in order to avoid any surprises.

Acked-by: Mark Rutland <mark.rutland@arm.com>
Signed-off-by: Dave Martin <Dave.Martin@arm.com>
[will: same change to write_sysreg_s]
Signed-off-by: Will Deacon <will.deacon@arm.com>
[mb: drop write_sysreg_s part]
Signed-off-by: Matthias Brugger <mbrugger@suse.com>
---
 arch/arm64/include/asm/sysreg.h |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/arch/arm64/include/asm/sysreg.h
+++ b/arch/arm64/include/asm/sysreg.h
@@ -252,7 +252,7 @@ static inline void config_sctlr_el1(u32
  * the "%x0" template means XZR.
  */
 #define write_sysreg(v, r) do {					\
-	u64 __val = (u64)v;					\
+	u64 __val = (u64)(v);					\
 	asm volatile("msr " __stringify(r) ", %x0"		\
 		     : : "rZ" (__val));				\
 } while (0)
